server/src/main/java/org/opensearch/cluster/service/MasterService.java

```diff
@@ -299,8 +299,15 @@ public static boolean assertNotMasterUpdateThread(String reason) {
     }
 
     private void runTasks(TaskInputs taskInputs) {
-        final String longSummary = logger.isTraceEnabled() ? taskInputs.taskSummaryGenerator.apply(true) : "";
-        final String shortSummary = taskInputs.taskSummaryGenerator.apply(false);
+        final String longSummary;
+        final String shortSummary;
+        if (taskInputs.updateTasks.size() <= 1000) {
```

Comment:
@Bukhtawar should we keep this limit to 100 or 50, that should be good enough to give a picture of tasks being processed.